packages/offix-client/src/OfflineClient.ts

```diff
@@ -111,46 +195,96 @@ export class OfflineClient implements ListenerProvider {
    *
    * @param options the MutationHelperOptions to create the mutation
    */
-  public offlineMutation<T = any, TVariables = OperationVariables>(
+  public async offlineMutation<T = any, TVariables = OperationVariables>(
     options: MutationHelperOptions<T, TVariables>): Promise<FetchResult<T>> {
     if (!this.apolloClient) {
       throw new Error("Apollo offline client not initialised before mutation called.");
     } else {
-      const mutationOptions = createMutationOptions<T, TVariables>(options);
-      mutationOptions.context.updateFunction = mutationOptions.update;
-      return this.apolloClient.mutate<T, TVariables>(
-        mutationOptions
-      );
+
+      const mutationOptions = this.createOfflineMutationOptions(options);
+
+      if (this.online) {
+        return this.apolloClient.mutate(mutationOptions);
+      } else {
+        const queueEntry = await this.queue.enqueueOperation(mutationOptions as unknown as MutationOptions);
+
+        const mutationPromise = new Promise(async (resolve, reject) => {
+          this.queue.assignHandlersToQueueEntry(queueEntry, resolve, reject);
+        });
+
+        throw new OfflineError(mutationPromise as any);
+      }
     }
   }
 
+  // TODO - does offix-cache actually need createMutationOptions?
```

Comment:
The `createMutationOptions` function comes from `offix-cache` right now. It isn't used anywhere else except within `OfflineClient.offlineMutate` I'm wondering is there any strong reason why `createMutationOptions` needs to be in `offix-cache`? Additionally, `createOfflineMutationOptions` in `OfflineClient` is a wrapper that adds extra things that are needed such as `conflictBase` and the `update` function from global cache update functions.

Basically the concern is that the original `createMutationOptions` in `offix-cache` kinda seems not fit for purpose to me and I also wonder why it's something we expose.

Reply:
Context for is that cache package is used externally by people who want to get cache updates out of the box. It has wider usage than Offix itself. However since with this change we kinda moving away from cache itself and cache updates are under control of user we might make choice to leave offix-cache unmaintained and bring some of the code inside. 

This will probably make things much cleaner, but once again this is something that we can do as next step once we go with scheduler and separate store.